laplace/curvature/backpack.py

```diff
@@ -154,6 +154,115 @@ def kron(self, X, y, **kwargs):
         raise NotImplementedError('Unavailable through Backpack.')
 
 
+class BackPackGP(BackPackInterface):
```

Comment:
Overall feedback for this class:
1. make sure the docstrings are consistent (in `k_b_b` you use another style currently)
2. make the function names potentially more verbose, e.g., `gp` --> `gp_quantities`, `k_b_b` -> `kernel_batch` maybe?
3. why do we need `k_b_b` and `k_star_star`? Can they somehow be collapsed together?
4. Is the `lambdas` function redundant with something else in the code and can be combined somehow?

Reply:
1. nice catch, done
2. agreed, done
3. I thought about this, however, was not able to come up with a single function (without too many if-else branches) that could handle both cases. The difference between the two is how we handle the batch dimension: in `K_batch` we "collapse" it, while in `K_star` we want to keep the batch dimension as it is. 
4. you're right, there was duplication with `_get_full_ggn` method in `GGNInterface`. I removed `lambdas` function now and added `H_log_likelihood` in `CurvatureInterface` instead